### PR TITLE
clean up non-throwable exception

### PR DESCRIPTION
**Pull request by imakhalova**
Some methods had throw in signature for exceptions that they don't throw them

### CODE REVIEW

File: src/src/com/microsoft/aad/adal/AuthenticationActivity.java [hunk 1]
```diff
@@ -886,127 +867,98 @@ private void appendAppUIDToAccount(StorageHelper cryptoHelper, Account account)
...
-        private void setAccount(final TokenTaskResult result) throws InvalidKeyException,
-                InvalidKeySpecException, InvalidAlgorithmParameterException,
-                IllegalBlockSizeException, BadPaddingException, IOException {
+        private void setAccount(final TokenTaskResult result)
+                throws GeneralSecurityException, UnsupportedEncodingException {
             // TODO Add token logging
             // TODO update for new cache logic
 
             // Authenticator sets the account here and stores the tokens.
-            try {
```
Comment (imakhalova) [inline src/src/com/microsoft/aad/adal/AuthenticationActivity.java]: I just removed try catch block from here...but this diff tool can't analyze it...

File: src/src/com/microsoft/aad/adal/StorageHelper.java [hunk 2]
```diff
@@ -384,6 +298,68 @@ private char getEncodeVersionLengthPrefix() {
...
 
+    /**
+     * encrypt text with current key based on API level
+     *
+     * @param clearText
+     * @return
+     * @throws GeneralSecurityException
+     * @throws UnsupportedEncodingException
+     */
+    public String encrypt(String clearText)
```
Comment (imakhalova) [inline src/src/com/microsoft/aad/adal/StorageHelper.java]: just changed the signature of this function

File: src/src/com/microsoft/aad/adal/StorageHelper.java [hunk 3]
```diff
@@ -530,6 +500,7 @@ private synchronized KeyPair getKeyPairFromAndroidKeyStore() throws GeneralSecur
     
     @TargetApi(18)
     private Object getKeyPairGeneratorSpec(X500Principal subj, Date start, Date end) {
+        // TODO: use real Spec.Builder instead of reflective methods
```
Comment (RandalliLama) [inline src/src/com/microsoft/aad/adal/StorageHelper.java]: Did you file an issue for this?

Comment (RandalliLama) [inline src/src/com/microsoft/aad/adal/StorageHelper.java]: @imakhalova ^

**Comment by RandalliLama** [comment]
I have only one comment.  Otherwise :shipit:

File: src/src/com/microsoft/aad/adal/StorageHelper.java (hunk 3 above)
Comment (weijjia) [inline src/src/com/microsoft/aad/adal/StorageHelper.java]: I filed an issue on this. https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/533

**Comment by weijjia** [comment]
@imakhalova I need you to resolve the conflicts, and then I can go ahead and merge it.

File: src/src/com/microsoft/aad/adal/StorageHelper.java (hunk 3 above)
Comment (imakhalova) [inline src/src/com/microsoft/aad/adal/StorageHelper.java]: Thanks @weijjia  for filing issue.
I don't really know the history why it uses reflective methods for KeyPairGeneratorSpec.Builder, I guess there were historical reasons behind (some problems on old versions or something else), and I think we need to know it before we do the change...

**Comment by weijjia** [comment]
@imakhalova I saw another problems in test, you change the testScenario_LoginHint_IdToken_Different context.acquireToken(testActivity, "resource", "clientid", "redirectUri", idtoken.upn, callback); take in the TestIdToken, but did you declare it anywhere? It's reporting error.

**Comment by imakhalova** [comment]
Got it, it was merge conflict and I didn't notice renamed variable...fixed.

**Pull request merged**